validator_client/src/validator_store.rs

```diff
@@ -154,15 +219,39 @@ impl<T: SlotClock + 'static, E: EthSpec> ValidatorStore<T, E> {
         validator_pubkey: &PublicKey,
         mut block: BeaconBlock<E>,
     ) -> Option<BeaconBlock<E>> {
-        // TODO: check for slashing.
-        self.validators
-            .read()
-            .get(validator_pubkey)
-            .and_then(|validator_dir| {
-                let voting_keypair = validator_dir.voting_keypair.as_ref()?;
-                block.sign(&voting_keypair.sk, &self.fork()?, &self.spec);
-                Some(block)
-            })
+        let validators = self.validators.read();
+
+        // Retrieving the corresponding ValidatorDir
+        let validator = match validators.get(validator_pubkey) {
+            Some(validator) => validator,
+            None => return None,
+        };
+
+        if validator.block_slashing_protection.is_none() {
+            error!(
+                self.log,
+                "Validator does not have block slashing protection";
+                "action" => "refused to produce block",
+                "pubkey" => format!("{:?}", &validator.voting_keypair.pk),
+            )
+        }
+
+        // Checking for slashing conditions
+        let is_slashing_free = validator
+            .block_slashing_protection
+            .as_ref()?
+            .try_lock()?
```

Comment:
What should we do here if it fail? Need to check upstream which behaviour we want?